tensorflow/training/docker/2.14/py3/cu118/Dockerfile.gpu

```diff
@@ -0,0 +1,406 @@
+#  ____                   __  __       _
+# / ___|  __ _  __ _  ___|  \/  | __ _| | _____ _ __
+# \___ \ / _` |/ _` |/ _ \ |\/| |/ _` | |/ / _ \ '__|
+#  ___) | (_| | (_| |  __/ |  | | (_| |   <  __/ |
+# |____/ \__,_|\__, |\___|_|  |_|\__,_|_|\_\___|_|
+#              |___/
+#  ___                              ____           _
+# |_ _|_ __ ___   __ _  __ _  ___  |  _ \ ___  ___(_)_ __   ___
+#  | || '_ ` _ \ / _` |/ _` |/ _ \ | |_) / _ \/ __| | '_ \ / _ \
+#  | || | | | | | (_| | (_| |  __/ |  _ <  __/ (__| | |_) |  __/
+# |___|_| |_| |_|\__,_|\__, |\___| |_| \_\___|\___|_| .__/ \___|
+#                      |___/                        |_|
+#################################################################
+
+FROM common AS sagemaker
+
+# Install SageMaker Profiler binary
+ARG SMP_URL=https://smppy.s3.amazonaws.com/tensorflow/cu118/smprof-0.3.332-cp310-cp310-linux_x86_64.whl
+RUN apt-get update  \
+ && apt-get install -y libunwind-dev  \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean \
+ && ${PYTHON} -m pip install --no-cache-dir ${SMP_URL}
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+ARG TF_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow/r2.14_aws/gpu/2024-01-23-01-49/tensorflow_gpu-2.14.1-cp310-cp310-linux_x86_64.whl
+
+# sagemaker-specific environment variable
+ENV SAGEMAKER_TRAINING_MODULE sagemaker_tensorflow_container.training:main
+
+# Need to install wheel before we can fix the pyyaml issue below
+RUN pip install wheel
+
+# https://github.com/yaml/pyyaml/issues/601
+# PyYaml less than 6.0.1 failes to build with cython v3 and above.
+# tf-models-official uses older versions, breaking the install.
+# going to install the older pyyaml and cython to get tfd-models-official
+# the sagemaker package will revert pyyaml back to 6 for its requirement
+# and this is fine since sagemaker is more important than the models and
+# the models still work on pyyaml 6 in this context.
+RUN pip install "cython<3" "pyyaml<6" --no-build-isolation
+
+# https://github.com/tensorflow/models/issues/9267
+# tf-models does not respect existing installations of TF and always installs open source TF
+RUN ${PIP} install --no-cache-dir -U \
+    tf-models-official==2.14.2 \
+    tensorflow-text==2.14.0 \
+ && ${PIP} uninstall -y tensorflow tensorflow-gpu \
+ && ${PIP} install --no-cache-dir -U \
+    ${TF_URL} \
+    "tensorflow-io>=0.33.0,<0.34.0" \
+    tensorflow-datasets
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    numba \
+    bokeh \
+    imageio \
+    opencv-python \
+    plotly \
+    seaborn \
+    shap
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    "sagemaker<3" \
+    sagemaker-experiments==0.* \
+    sagemaker-tensorflow-training \
+    sagemaker-training \
+    "sagemaker-studio-analytics-extension<1" \
+    "sparkmagic<1" \
+    "sagemaker-studio-sparkmagic-lib<1" \
+    smclarify
+
+# install sagemaker-tensorflow from source
```

Comment:
do we need to install the package from source? is pip install supported?

Reply:
No, it is available on pypi. It's just a carryover from the TF 2.13 dockerfile. Will try using the pypi wheel directly instead of installing from source, and provide an update.

Reply:
Using the wheel from pypi caused a runtime error as documented in the new comment explaining why we install this package from source.